Mathlib/Data/List/Basic.lean

```diff
@@ -395,6 +395,14 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
 @[deprecated] alias append_right_cancel := append_cancel_right -- deprecated since 2024-01-18
 #align list.append_right_cancel List.append_cancel_right
 
+@[simp] theorem append_left_cancel_nil_left {x y : List α} : x ++ y = x ↔ y = [] := by
+  constructor <;> intro h
+  · apply List.append_cancel_left (as := x); simp [h]
```

Comment:
Can you write this as two lines, or better, combine into one? Here's a Lean trick: `by simp [h]` creates a proof term again, which you can use in the apply - i.e., writing `apply List.append_cancel_left (as := x) (by simp [h])`.
Then you can replace `apply` by `exact`.